src/ansys/aedt/core/visualization/post/post_3dlayout.py

```diff
@@ -43,43 +46,113 @@ class PostProcessor3DLayout(PostProcessor3D):
+                    file_net = i
+                    break
+            with open(file_net, "r") as f:
+                file_net_text = f.read()
+                match = re.search(r"B_NET_CLASSIFICATION\s+(.*?)\s+E_NET_CLASSIFICATION", file_net_text, re.DOTALL)
+                nets = []
+                for i in match.group(1).split("\n"):
+                    net_name = i.lstrip(" ").split(" ")[1]
+                    nets.append(net_name)
+
+        edbapp = self._app.modeler.edb
+        net_per_layer_names = {i: [] for i in edbapp.stackup.signal_layers.keys()}
+
+        for net_name in nets:
+            net_obj = edbapp.nets[net_name]
+            layer_names = [i.layer_name for i in net_obj.primitives]
+            layer_names = list(set(layer_names))
+            for i in layer_names:
+                net_per_layer_names[i].append(net_name)
+
+        power_loss_per_layer = []
+        operations = []
+        for layer_name, net_names in net_per_layer_names.items():
+            if not net_names:
+                continue
+            thickness = edbapp.stackup[layer_name].thickness
+            for net_name in net_names:
+                assignment = f"{layer_name}_{net_name}"
+                operations.extend(
+                    [
+                        "Fundamental_Quantity('P')",
+                        f"EnterSurface('{assignment}')",
+                        "Operation('SurfaceValue')",
+                        "Operation('Integrate')",
+                    ]
+                )
+
+                operations.extend([f"Scalar_Constant({thickness})", "Operation('*')"])
+
+                solution_type = ["DC Fields"] if settings.aedt_version < "2025.1" else ["DCIR Fields"]
+                my_expression = {
+                    "name": f"Power_{layer_name}",
+                    "description": "Power Density",
+                    "design_type": ["HFSS 3D Layout Design"],
+                    "fields_type": solution_type,
+                    "solution_type": "",
+                    "primary_sweep": "",
+                    "assignment": "",
+                    "assignment_type": ["Surface"],
+                    "operations": operations,
+                    "report": ["Data Table", "Rectangular Plot"],
+                }
+                if self._app.post.fields_calculator.is_expression_defined(my_expression["name"]):
+                    self._app.post.fields_calculator.delete_expression(my_expression["name"])
+                self._app.post.fields_calculator.add_expression(my_expression, "")
+
+                loss = self._app.post.fields_calculator.evaluate(my_expression["name"], solution, intrinsics={})
+
+                power_loss_per_layer.append({"layer": layer_name, "net": net_name, "loss": float(loss)})
+
+        return power_loss_per_layer
 
     @pyaedt_function_handler()
-    def compute_power_by_layer(self, layers=None, nets=None, solution=None):
+    def compute_power_by_layer(self, solution=None):
```

Comment:
I would have the possibility to filter by layer or net in any case (even if now is much faster). On big board could simplify the post-processing